server/lobby.js

```diff
@@ -446,9 +449,36 @@ class Lobby {
                 return;
             }
 
-            game.selectDeck(socket.user.username, deck);
+            this.cardRepository.getCards(false, (err, cards) => {
+                if(err) {
+                    return;
+                }
 
-            this.sendGameState(game);
+                this.cardRepository.getPacks((err, packs) => {
+                    if(err) {
+                        return;
+                    }
+
+                    _.each(deck.plotCards, plot => {
+                        plot.card = cards[plot.card.code];
+                    });
+
+                    _.each(deck.drawCards, draw => {
+                        draw.card = cards[draw.card.code];
+                    });
+
+                    if(deck.agenda) {
+                        deck.agenda = cards[deck.agenda.code];
+                    }
+
+                    let validation = validateDeck(deck, packs);
+                    deck.status = validation.status;
```

Comment:
Again on the nitpicky side, but it would be nice if this method read more like:
```javascript
this.loadDeck(deckId, deck => { // or some method on deckRepository
    game.selectDeck(socket.user.username, deck);
    this.sendGameState(game);
});
```

Instead of the 3 nested callbacks it is currently:
```javascript
this.deckRepository.getById(deckId, deck => {
    this.cardRepo.getCards(cards => {
        this.cardRepo.getPacks(packs => {
            // do a bunch of stuff preparing the deck
            // FINALLY actually select the deck and send data
        });
    });
});